packages/python-packages/protocol-stub-generator/protocol/protocol_models.py

```diff
@@ -671,6 +794,398 @@ def to_json(self):
         return obj_dict
 
 
+def request_builder(
```

Comment:
there's a lot of code in this function body. I think we need to refactor out parts so that it's easier to follow.